src/platform/packages/shared/kbn-esql-ast/src/parser/cst_to_ast_converter.ts

```diff
@@ -1242,12 +1242,164 @@ export class CstToAstConverter {
 
   // ------------------------------------------------------------------- RERANK
 
+  /**
+   * Supports two syntax forms:
+   * - RERANK "query" ON fields WITH {options}
+   * - RERANK target = "query" ON fields WITH {options}
+   */
   private fromRerankCommand(ctx: cst.RerankCommandContext): ast.ESQLAstRerankCommand {
-    const command = this.createCommand<'rerank', ast.ESQLAstRerankCommand>('rerank', ctx, {});
+    const command = this.createCommand<'rerank', ast.ESQLAstRerankCommand>('rerank', ctx);
+
+    this.parseRerankQuery(ctx, command);
+    this.parseRerankFields(ctx, command);
+    this.parseRerankWithParameters(ctx, command);
 
     return command;
   }
 
+  /**
+   * Parses the query text and optional target field assignment.
+   * Handles: RERANK [target =] "query" ...
+   */
+  private parseRerankQuery(ctx: cst.RerankCommandContext, command: ast.ESQLAstRerankCommand): void {
+    if (!ctx._queryText) {
+      return;
+    }
+
+    const queryText = this.fromConstant(ctx._queryText);
```

Comment:
Comments in the PR https://github.com/elastic/kibana/issues/217285 suggest that the query part can be parametric but the grammar says it is a constant

Reply:
yes this is outdated, we follow our grammar